### DISCUSSION TITLE
Rom report: where do the no path and hidden symbols come from

### DISCUSSION THREAD

**Discussion by konglualualu** [zephyrproject-rtos/zephyr#56958] [Q&A] [answered]
I try “west build -t rom_report” to see my project rom report，but found out that a large proportion of the size is hidden or no path.
Where do these symbols come from? How can I optimize it？
![1](https://user-images.githubusercontent.com/9513819/232665811-aea15459-edc2-4c6a-bd2e-ba1a4d09a349.png)
![2](https://user-images.githubusercontent.com/9513819/232664626-457269ce-842c-441a-8d44-e9f26c54701e.png)

**Comment by DerekSnell** [comment, marked as answer]
Hi @konglualualu ,

I do not know the details about the rom_report tool. But the linker also generates a file that summarizes the memory usage. Typically this is a file located at <build folder>/zephyr/zephyr.map.

Perhaps reviewing that file will help understand the hidden items in this report.  Best regards

**Reply by konglualualu** [reply]
thanks @DerekSnell , I originally thought that the rom_report tool can easily see the proportion distribution of each module or .c in the code space, but for now, map file do seem to be the better way. thanks anyway.
